components/searchable_user_list/searchable_user_list.jsx

```diff
@@ -167,7 +167,20 @@ export default class SearchableUserList extends React.Component {
         let usersToDisplay;
 
         if (this.props.term || !this.props.users) {
-            usersToDisplay = this.props.users;
+            usersToDisplay = [...this.props.users];
+
+            if (this.props.term && usersToDisplay) {
+                const searchTerm = this.props.term.replace(/^@/, '');
+                usersToDisplay.sort((a, b) => {
+                    if (a.username.startsWith(searchTerm)) {
```

Comment:
This may not work right depending on how the browser does sorting. First off, if both `a` and `b` start with the searchTerm, `a` will always come first even if it's alphabetically after `b`. Second, returning 0 here assumes that the browser will keep `a` and `b` in the correct order that they were in before, but that's not guaranteed so it would be best to assume that it won't do that.

Instead, we should include also sort on the name if either both terms start with the search term or neither do like https://github.com/mattermost/mattermost-webapp/blob/master/components/suggestion/emoticon_provider.jsx#L97